plugins/woocommerce/client/blocks/assets/js/base/components/cart-checkout/form/use-form-fields.ts

```diff
@@ -72,10 +72,23 @@ export const useFormFields = < T extends keyof FormFields >(
 				}
 			}
 			if ( hasSchemaRules( defaultConfig, 'hidden' ) ) {
-				const schema = {
-					type: 'object',
-					properties: defaultConfig.hidden,
-				};
+				let schema = {};
+				if (
+					Object.keys( defaultConfig.hidden ).some(
+						( key ) =>
+							key === 'cart' ||
+							key === 'checkout' ||
+							key === 'customer'
+					)
+				) {
+					schema = {
+						type: 'object',
+						properties: defaultConfig.hidden,
+					};
+				} else {
+					schema = defaultConfig.hidden;
+				}
```

Comment:
Minor: since this logic is repeated 2 times, you can consider extracting it to a function, could offer more clarity into what we're doing here.

Reply:
I think we could abstract the `Object.keys` condition to make the code more digestible but I'm in favour of being explicit about the rest of the code here and leaving it. That said, not a super strong opinion on this.

Reply:
I didn't feel like it's worth abstracting this yet, but maybe you're right.